src/Controllers/ContentJsonController.php

```diff
@@ -550,4 +537,54 @@ public function countLessonsAndAssignments($contentId)
     {
         return $this->contentService->countLessonsAndAssignments($contentId);
     }
+
+    /**
+ * @param Request $request
+ * @param string $group_by
+ * @param mixed $required_fields
+ * @param mixed $included_fields
+ * @return array
+ */
+    private function extractFields(Request $request, string $group_by, mixed $required_fields, mixed $included_fields)
+    : array {
+        if ($request->get('tabs', false)) {
+            $tabs = $request->get('tabs');
+            if (!is_array($request->get('tabs'))) {
+                $tabs = [$request->get('tabs')];
+            }
+            foreach ($tabs as $tab) {
+                $extra = explode(',', $tab);
+                if ($extra['0'] == 'group_by') {
+                    $group_by = $extra['1'];
+                }
+                if ($extra['0'] == 'duration') {
+                    $required_fields[] = 'length_in_seconds,'.$extra[1].',integer,'.$extra[2].',video';
+                }
+                if ($extra['0'] == 'length_in_seconds' || $extra['0'] == 'topic') {
+                    $required_fields[] = $tab;
+                }
+            }
+        }
+
+        if ($request->has('title') && ($group_by == 'artist' || $group_by == 'style')) {
+            $required_fields[] = $group_by.',%'.$request->get('title').'%,string,like';
+        } elseif ($request->has('title') && $group_by == 'instructor') {
+            $instructors =
+                $this->contentService->getWhereTypeInAndStatusAndField(
+                    ['instructor'],
+                    ['published'],
```

Comment:
the status is expected to be a string
```suggestion
                    'published',
```